commons/src/main/java/org/infinispan/commons/util/CollectionFactory.java

```diff
@@ -101,8 +103,7 @@ public static int computeCapacity(int expectedSize, float loadFactor) {
    }
 
    public static <K, V> ConcurrentMap<K, V> makeBoundedConcurrentMap(int maxSize) {
-      return new BoundedEquivalentConcurrentHashMapV8<>(maxSize,
-            AnyEquivalence.getInstance(), AnyEquivalence.getInstance());
+      return new CaffeineConcurrentMap<>(Caffeine.newBuilder().maximumSize(maxSize).build());
```

Comment:
This PR doesn't actually remove the BECHMV8 classes, but they probably should go.
Also we need to document this change in both the user guide and in the upgrading doc